### PR TITLE
Fix idling aircraft on map Intervention

### PR DESCRIPTION
**Pull request by obrakmann**

### CODE REVIEW

**Comment by pchote** [comment]
Do we want this for the prep branch?

File: mods/ra/maps/intervention/rules.yaml [hunk 1]
```diff
@@ -76,6 +76,7 @@ MIG:
 		Prerequisites: ~afld
 	AmmoPool:
 		Ammo: 2
+	-Rearmable:
```
Comment (abcdefg30) [inline mods/ra/maps/intervention/rules.yaml]: The unit is still buildable, so not sure if that is a good idea. If we have to do such changes we'll need to clone the unit. (Btw, do you think the `LeaveMap` `IdleBehavior` can help here?)

**Review by abcdefg30 — COMMENTED** [review]
Odd, this must have regressed again. Imho yes, on the milestone.

**Review by abcdefg30 — DISMISSED** [review]
Confirmed that the migs leave the map again. :+1: with the comment addressed.

**Comment by tovl** [comment]
IMO this is getting pretty hacky. The real issue is that `ReturnToBase` cancels the whole queue, which is against our own guidelines for how activities should behave.

The proper solution is to remove the activity cancelling from `ReturnToBase` entirely and let `FlyAttack` be solely responsible for breaking out of a potential infinite loop. This can be done by checking whether the branch of `FlyAttack.Tick` that calls `ReturnToBase` is entered two ticks in a row and going to `NextActivity` if that is the case.

**Comment by pchote** [comment]
That sounds like a reasonable approach, but we will need to be careful that it doesn't introduce behaviour changes for other uses (like the Lua API).

File: mods/ra/maps/intervention/rules.yaml (hunk 1 above)
Comment (obrakmann) [reply mods/ra/maps/intervention/rules.yaml]: I like the IdleBehavior suggestion. I lost track of that and wasn't aware that it had already been merged.

**Review by abcdefg30 — APPROVED** [review]
:white_check_mark: Fixes the issue without the controversial code changes.

**Review by reaperrr — APPROVED** [review]

**Comment by reaperrr** [comment]
https://github.com/OpenRA/OpenRA/wiki/Changelog/_compare/6f97f1843228f59effd4c0ecb78f103ca50a8b95...641bd8d22e105909b29b6418494949d93296d8d0

e48732c

**Pull request merged**